geode-web-management/src/main/java/org/apache/geode/management/internal/rest/ManagementLoggingFilter.java

```diff
@@ -15,32 +15,71 @@
 
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
 import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
 
-import org.springframework.web.filter.AbstractRequestLoggingFilter;
+import org.springframework.web.filter.OncePerRequestFilter;
+import org.springframework.web.util.ContentCachingRequestWrapper;
+import org.springframework.web.util.ContentCachingResponseWrapper;
 
-public class ManagementLoggingFilter extends AbstractRequestLoggingFilter {
+public class ManagementLoggingFilter extends OncePerRequestFilter {
 
   // Because someone is going to want to disable this.
   private static final Boolean ENABLE_REQUEST_LOGGING =
       Boolean.parseBoolean(System.getProperty("geode.management.request.logging", "true"));
 
-  public ManagementLoggingFilter() {
-    super.setIncludeQueryString(true);
-    super.setIncludePayload(true);
-    super.setMaxPayloadLength(1000);
-    super.setAfterMessagePrefix("Management request: [");
-  }
+  private static int MAX_PAYLOAD_LENGTH = 1000;
 
   @Override
-  protected void beforeRequest(HttpServletRequest request, String message) {
-    // No logging here - this would not display the payload
+  protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response,
+      FilterChain filterChain) throws ServletException, IOException {
+    if (!ENABLE_REQUEST_LOGGING) {
+      return;
+    }
+
+    // We can not log request payload before making the actual request because then the InputStream
+    // would be consumed and cannot be read again by the actual processing/server.
+    ContentCachingRequestWrapper wrappedRequest = new ContentCachingRequestWrapper(request);
+    ContentCachingResponseWrapper wrappedResponse = new ContentCachingResponseWrapper(response);
+    // performs the actual request before logging
+    filterChain.doFilter(wrappedRequest, wrappedResponse);
+
+    // log after the request has been made and ContentCachingRequestWrapper has cached the request
+    // payload
+    String requestPattern = "Management Request: %s[url=%s]; user=%s; payload=%s";
+    String requestUrl = request.getRequestURI();
+    if (request.getQueryString() != null) {
+      requestUrl = requestUrl + "?" + request.getQueryString();
+    }
+    String payload = getContentAsString(wrappedRequest.getContentAsByteArray(),
+        wrappedRequest.getCharacterEncoding());
+    logger.info(String.format(requestPattern, request.getMethod(), requestUrl,
+        request.getRemoteUser(), payload));
+
+
+    // construct the response message
+    String responsePattern = "Management Response: Status=%s; response=%s";
+    payload = getContentAsString(wrappedResponse.getContentAsByteArray(),
+        wrappedResponse.getCharacterEncoding());
+    logger.info(String.format(responsePattern, response.getStatus(), payload));
```

Comment:
I see this payload ends up with a bunch of newlines. Can you strip those out so we only log a single message line? Having multiple log lines (for a single entry) will mess up any log scraping that someone might do.

Reply:
done